premis_event_service/models.py

```diff
@@ -93,20 +93,45 @@ def search(self, **kwargs):
         outcome = kwargs.get('event_outcome')
         event_type = kwargs.get('event_type')
         linked_object_id = kwargs.get('linked_object_id')
+        min_ordinal = kwargs.get('min_ordinal')
 
-        events = self.get_queryset().order_by('-event_date_time')
+        events = self.get_queryset().order_by('-ordinal')
 
-        # Filter based on the supplied the arguments.
-        events = events.filter(event_date_time__gte=start_date) if start_date else events
-        events = events.filter(event_date_time__lte=end_date) if end_date else events
-        events = events.filter(event_outcome=outcome) if outcome else events
-        events = events.filter(event_type=event_type) if event_type else events
+        # This shouldn't happen in practice, but this conditional prevents an
+        # unfiltered query from getting through this function. The stakes are
+        # reasonably high, as it becomes a very, very slow, very ,very resource
+        # intensive query if it does happen.
+        if any((start_date, end_date, outcome, event_type, linked_object_id)):
```

Comment:
It's here purely to differentiate between unfiltered and filtered queries. Later on, the `link_object_id` is used as a filter, if it's provided.